src/io/filewatcher.rs

```diff
@@ -53,17 +56,33 @@ impl FileWatcher {
         // Reset if file has been remade (i.e. is shorter) and update state
         if (meta.len() as usize) < self.last_size {
             start_idx = 0;
+            self.file = OpenOptions::new()
+                .read(true)
+                .write(false)
+                .open(&self.file_path)
+                .context("Failed to reopen file.")?;
         }
 
         self.last_size = meta.len() as usize;
-
+        
         // Get new file contents
-        let mut file =
-            FileWatcher::open_file(&self.file_path).context("Failed to reopen log file")?;
+        // self.file.seek(SeekFrom::Start(start_idx as u64))?;
```

Comment:
You can get rid of these statements rather than just commenting them out, I don't think we will be needing them again.